python/golem_task_api/client.py

```diff
@@ -37,18 +38,35 @@
 CONNECTION_TIMEOUT = 5.0  # seconds
 
 
-async def wait_until_socket_open(
+# grpclib.health.v1.HealthCheckResponse._HEALTHCHECKRESPONSE_SERVINGSTATUS
+class ServingStatus(Enum):
+    UNKNOWN = 0
+    SERVING = 1
+    NOT_SERVING = 2
+    SERVICE_UNKNOWN = 3
+
+
+async def wait_for_channel(
         host: str,
         port: int,
         timeout: float = CONNECTION_TIMEOUT
-) -> None:
+) -> Channel:
+    request = HealthCheckRequest()
+    request.service = ''  # empty service name for a general check
+
     deadline = time.time() + timeout
     while time.time() < deadline:
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        with contextlib.closing(sock):
-            if sock.connect_ex((host, port)) == 0:
-                return
+        channel = Channel(host, port, loop=asyncio.get_event_loop())
+        client = HealthStub(channel)
+        try:
+            response = await client.Check(request)
+            status = ServingStatus(response.status)
+            if status is ServingStatus.SERVING:
```

Comment:
After reading the code for `grpclib.health.service.Health` it seems to me that the returned status could be ignored because `OVERALL` check always returns `SERVING`. So the fact that service responded would be enough of a check here. Correct me if I'm wrong.

Reply:
I mean we don't pass any arguments for `Health()` in `AppServer` constructor so there are no actual checks to perform, right?

Reply:
That's true - the check is not necessary. But IMO leaving the check makes the intent clearer and does not require from the reader to possess that knowledge.

Reply:
OK, fair enough. We might also enable adding custom health checks by app developers some time later.